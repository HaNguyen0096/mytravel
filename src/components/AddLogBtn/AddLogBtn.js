import { Button } from 'antd'
import React, { Component } from 'react'
import { Link } from 'react-router-dom'

export default class AddLogBtn extends Component {
  render(){
    const path = `/addlog`
    
    return (
      <div>
        <Button type='primary'><Link to={path}>Add New Log</Link></Button>
      </div>
    )
  }
}