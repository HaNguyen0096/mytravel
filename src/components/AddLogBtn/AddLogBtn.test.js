import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import AddLogBtn from './AddLogBtn'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('AddLogBtn component', () => {

  it('renders AddLogBtn without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
                      <AddLogBtn />
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI for AddLogBtn as expected', () => {
    const tree = renderer
      .create(<BrowserRouter>
                <AddLogBtn />
              </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders AddLogBtn by default', () => {
    const wrapper = shallow(<AddLogBtn />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})