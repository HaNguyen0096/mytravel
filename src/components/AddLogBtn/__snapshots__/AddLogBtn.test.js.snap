// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddLogBtn component renders AddLogBtn by default 1`] = `
<div
  className="addLogBtn"
>
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    loading={false}
    type="primary"
  >
    <Link
      to="/addlog"
    >
      Add New Log
    </Link>
  </Button>
</div>
`;

exports[`AddLogBtn component renders the UI for AddLogBtn as expected 1`] = `
<div
  className="addLogBtn"
>
  <button
    className="ant-btn ant-btn-primary"
    onClick={[Function]}
    type="button"
  >
    <a
      href="/addlog"
      onClick={[Function]}
    >
      Add New Log
    </a>
  </button>
</div>
`;
